lib/id3c/cli/command/etl/fhir.py

```diff
@@ -176,8 +176,9 @@ def process_encounter_bundle_entry(db: DatabaseSession, bundle: Bundle, entry: B
     LOG.debug(f"Processing Encounter Resource «{entry.fullUrl}».")
 
     related_resources = extract_related_resources(bundle, entry)
-
-    encounter = process_encounter(db, resource, related_resources)
+    immunization_resources = extract_immunization_resources(bundle)
+    
+    encounter = process_encounter(db, resource, related_resources, immunization_resources)
```

Comment:
Is the processing of immunization resources split out from "related_resources" here because related resources is supposed to be encounter level, and immunization is supposed to be patient level?

Reply:
> Is the processing of immunization resources split out from "related_resources" here because related resources is supposed to be encounter level, and immunization is supposed to be patient level?

Exactly, immunization resource entries are only linked to patients in this case. The would only be linked to an encounter if the encounter was the actual immunization event. In our case these are "historical immunizations" which do not reference a specific encounter: https://www.hl7.org/fhir/immunization-example-historical.json.html